libmuscle/python/libmuscle/communicator.py

```diff
@@ -60,6 +62,48 @@ def __init__(self, timestamp: float, next_timestamp: Optional[float] = None,
         self.settings = settings
 
 
+class RecvTimeoutHandler(TimeoutHandler):
```

Comment:
Yes makes sense. I've had to include a factory class as well to make this work, but it simplifies `communicator.py` so that's worth it IMO!

Note: we can't set it in `Communicator.__init__` because the timeout needs to have the settings received. To have the settings, we must have registered with the muscle manager. And `Instance._register` needs to have an initialized `Communicator`

Reply:
Ah, good point about __init__, I meant during initialisation in general, it's similar to setting the log levels.